lib/modules/dosomething/dosomething_api/resources/kudos_resource.inc

```diff
@@ -145,9 +148,17 @@ function _kudos_resource_access($op) {
   }
 
   if ($op === 'create' || $op === 'delete') {
-    if (!user_is_anonymous()) {
-      return TRUE;
+    // Only auth'd users should be giving Kudos
+    if (user_is_anonymous()) {
+      return FALSE;
+    }
+
+    // We need to make sure people are only creating Kudos for themselves
+    if ($op === 'create' && $user->uid !== $args[1]) {
```

Comment:
Are there scenarios like the mobile app, where the "logged-in" user is a generic auth'd user for the sake of allowing access to certain things, but it's not _actually_ the current user... that data is just getting passed around? I feel that's the scenario it will break on.